spa/src/api/rest-api.ts

```diff
@@ -0,0 +1,75 @@
+export function createRestApi() {
+
+	let gitHubAccessToken: string | undefined;
+	let gitHubRefreshToken: string | undefined;
+
+	return {
+		storeGitHubToken,
+		isTokenStillValid,
+		generateOAuthUrl,
+		fetchInstalledOrganisations
+	};
+
+	function storeGitHubToken(tokens: { gitHubAccessToken: string; githubRefreshToken?: string }) {
+		gitHubAccessToken = tokens.gitHubAccessToken;
+		gitHubRefreshToken = tokens.githubRefreshToken;
+	}
+
+	async function isTokenStillValid() {
+
+		try {
+			const resp = await fetch("https://api.github.com/user", {
+				headers: {
+					Authorization: `Bearer ${gitHubAccessToken}`
+				}
+			});
+			if (!resp.ok) {
+				console.log("Token invalid");
+				return false;
+			} else {
+				console.log("Token still valid");
+				return true;
+			}
+		} catch (e) {
+			console.log("Some network error on the request", e);
+			return false;
+		}
+	}
+
+	async function generateOAuthUrl() {
+		return "";
+	}
+
+	async function fetchInstalledOrganisations() {
+
+		try {
+
+			const jiraJwt = await getNewJiraToken();
+
+			const resp = await fetch("/rest/app/cloud/organizations", {
+				headers: {
+					Authorization: `Bearer ${jiraJwt}###${gitHubAccessToken}`
```

Comment:
We need both `gitHubToken` and `JiraJwtToken` in api request now. And I don't trust any other header than `Authorization`. <--- I think some switch/router in the middle of the network will all be aware of NOT to log the value of `Authorization`, but if we use `Some-random-header-key`, those router/switch might log them in the log.

Reply:
But I am not too sure, I am open for suggestions.